src/Interpreters/maskSensitiveInfoInQueryForLogging.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <Parsers/IAST_fwd.h>
+#include <Interpreters/Context_fwd.h>
+
+
+namespace DB
+{
+
+/// Makes a version of a query without sensitive information (e.g. passwords) for logging.
+/// The parameter `parsed query` is allowed to be nullptr if the query cannot be parsed.
+String maskSensitiveInfoInQueryForLogging(const String & query, const ASTPtr & parsed_query, const ContextPtr & context);
+
+/// Makes a version of backup name without sensitive information (e.g. passwords) for logging.
+String maskSensitiveInfoInBackupNameForLogging(const String & backup_name, const ASTPtr & ast, const ContextPtr & context);
```

Comment:
In my opinion, it's better to keep this in Parsers folder (instead of Interpreters) as it deals mostly with ASTs searching & replacing, also its not easy to use things from Interpreter while building other libs like lib-parsers.

Reply:
That is what I tried to do first, however `maskSensitiveInfoInBackupNameForLogging()` uses some stuff from Interpreters and other dbms targets: `evaluateConstantExpression()`, `TableFunctionFactory`, `FormatFactory`.